### PR TITLE
Add went on leave view  for teacher and student

### LINKED ISSUE #267: API view: `Student` or `Teacher` went on leave
## Add log event types before creating serializers and views

Please add the following log event types to `StudentLogEventType`, `TeacherLogEventType` and `TeacherUnder18LogEventType` (they can be copied from `CoordinatorLogEventType`)

```py
    GONE_ON_LEAVE = "gone_on_leave", "Gone on leave"
    RETURNED_FROM_LEAVE = "returned_from_leave", "Returned from leave"
```

When a coordinator from dashboard or bot sends a request stating that a student or a teacher went on leave:

## Create log events

Type of `LogEvent`s to be created: `GONE_ON_LEAVE`

## Set statuses
`project_status` is to be set to `ON_LEAVE`.

> [!Important]
> Remember to update `status_since` to current time in UTC.

### PR DESCRIPTION
**Pull request by Auch-Auch**
Closes #267

### CODE REVIEW

File: api/processors/actions/student/went_on_leave.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        self._set_statuses()
+
+    def _set_statuses(self) -> None:
+        self.student.project_status = StudentProjectStatus.ON_LEAVE
+        self.student.status_since = self.timestamp
+        self.student.save()
+
+    def _create_log_events(self) -> None:
+        StudentLogEvent.objects.create(
+            student=self.student, type=StudentLogEventType.GONE_ON_LEAVE
```
Comment (Auch-Auch) [inline api/processors/actions/student/went_on_leave.py:26]: Note that group is None

Comment (lemontree210) [reply api/processors/actions/student/went_on_leave.py:26]: I think it makes sense. The student will either drop out of the group first or (unlikely, though?) just continue with the same group when they return.

File: api/processors/actions/__init__.py [hunk 2]
```diff
@@ -16,4 +18,7 @@
     "StudentTransferProcessor",
     "StudentMissedClassProcessor",
     "TeacherTransferProcessor",
+    "StudentWantOnLeave",
```
Comment (lemontree210) [inline api/processors/actions/__init__.py:21]:
```suggestion
    "StudentWentOnLeave",
```
or maybe something completely different should be here (or nothing?)

**Review by lemontree210 — APPROVED** [review]
Thank you! Just one typo, I think.

File: api/processors/actions/__init__.py (hunk 2 above)
Comment (Auch-Auch) [reply api/processors/actions/__init__.py]: Ah, it is just redundant default import, it is not needed here

**Pull request merged**